libmatemixer/matemixer-stream-control.c

```diff
@@ -284,6 +286,19 @@ mate_mixer_stream_control_init (MateMixerStreamControl *control)
     control->priv = mate_mixer_stream_control_get_instance_private (control);
 }
 
+static void
+mate_mixer_stream_control_dispose (GObject *object)
+{
+    MateMixerStreamControl *control;
+
+    control = MATE_MIXER_STREAM_CONTROL (object);
+    if (control->priv->stream != NULL)
+        g_object_remove_weak_pointer(G_OBJECT (control->priv->stream),
+                                     (gpointer *) &control->priv->stream);
```

Comment:
This could also be done in the existing `finalize()`, as it's not releasing a reference, but just avoiding the weak pointer from accessing this object's memory after destruction.

This is just a note, it's also fine as it is.